tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -0,0 +1,331 @@
+                if (r.length === 1) {
+                    r = "0" + r;
+                }
+
+                return r;
+            }
+
+            return date.getUTCFullYear()
+                + "-" + pad(date.getUTCMonth() + 1)
+                + "-" + pad(date.getUTCDate())
+                + "T" + pad(date.getUTCHours())
+                + ":" + pad(date.getUTCMinutes())
+                + ":" + pad(date.getUTCSeconds())
+                + "." + String((date.getUTCMilliseconds() / 1000).toFixed(3)).slice(2, 5)
+                + "Z";
+        }
+
+        function _createEnvelope(iKey:string, theType:string) {
+            let tags = {};
+            let type = "Browser";
+            tags[strAiDevice + "id"] = type[strToLowerCase]();
+            tags[strAiDevice + "type"] = type;
+            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
+            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.sv || appInsights.version);
+
+            return {
+                time: _getTime(),
+                iKey: iKey,
+                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
+                sampleRate: 100,
+                tags: tags,
+                data: {
+                    baseData: {
+                        ver: 2
+                    }
+                }
+            };
+        }
+
+        function _createInternal(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : Envelope = _createEnvelope(iKey, "Message");
+            let data = envelope.data;
+            data.baseType = "MessageData";
+            let baseData = data.baseData;
+
+            baseData.message = "AI (Internal): 99 message:\"" + (message + " (" + targetSrc + ")").replace(/\"/g, strEmpty) + "\"";
+            baseData.properties = {
+                endpoint: endpointUrl
+            };
+            
+            return envelope;
+        }
+
+        function _createException(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : Envelope = _createEnvelope(iKey, "Exception");
+            let data = envelope.data;
+            data.baseType = "ExceptionData";
+            data.baseData.exceptions = [{
+                typeName: "SDKLoadFailed",
+                message: message.replace(/\./g, "-"),  // Replacing '.' characters as it causes the portal to hide the start of the message in the summary
+                hasFullStack: false,
+                stack: message + "\nSnippet failed to load [" + targetSrc + "] -- Telemetry is disabled\nHelp Link: " + helpLink + "\nHost: " + (locn && locn.pathname || "_unknown_") + "\nEndpoint: " + endpointUrl,
+                parsedStack: []
+            }];
+
+            return envelope;
+        }
+    
+        // Assigning these to local variables allows them to be minified to save space:
+        let targetSrc = aiConfig.url || snipConfig.src;
```

Comment:
aiConfig.url here is really here for backwards compatibility.

Lets use by casting rather than defining a type with this.

`let targetSrc = (aiConfig as any)["url"] || snipConfig.src;